sql/catalog/lease: TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces failed [test deadlocks in LeaseAcquiredEvent testing knob]

sql/catalog/lease.TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16491529?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/16491529?buildTab=artifacts#/) on master @ [5907123373f975b4c4eb3695f0222c1912ed8b2a](https://github.com/cockroachdb/cockroach/commits/5907123373f975b4c4eb3695f0222c1912ed8b2a):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces (13m30s)

```
Stack: 

```
goroutine 220949 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2366 +0x30c
created by time.goFunc
	GOROOT/src/time/sleep.go:177 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/1e31bde6494190748168afe7a747530a/logTestRangefeedUpdatesHandledProperlyInTheFaceOfRaces2618326280
    test_log_scope.go:81: use -show-logs to present logs inline
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRangefeedUpdatesHandledProperlyInTheFaceOfRaces.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-41371

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -1577,17 +1577,20 @@ func (m *Manager) checkRangeFeedStatus(ctx context.Context) (forceRefresh bool) 
 // range feed progress / recovery, and supporting legacy expiry
 // based leases.
 func (m *Manager) RunBackgroundLeasingTask(ctx context.Context) {
-	_ = m.stopper.RunAsyncTask(ctx, "lease-refresher", func(ctx context.Context) {
-		refreshTimerDuration := LeaseDuration.Get(&m.storage.settings.SV)
-		renewalsDisabled := false
-		if refreshTimerDuration <= 0 {
+	renewalsDisabled := false
+	getRefreshTimerDuration := func() time.Duration {
+		if LeaseDuration.Get(&m.storage.settings.SV) <= 0 {
 			// Session based leasing still needs a refresh loop to expire
 			// leases, so we will execute that without any renewals.
-			refreshTimerDuration = time.Millisecond * 200
 			renewalsDisabled = true
+			return 200 * time.Millisecond
 		} else {
-			refreshTimerDuration = m.storage.jitteredLeaseDuration()
+			renewalsDisabled = false
+			return m.storage.jitteredLeaseDuration()
 		}
+	}
+	_ = m.stopper.RunAsyncTask(ctx, "lease-refresher", func(ctx context.Context) {
+		refreshTimerDuration := getRefreshTimerDuration()
 		var refreshTimer timeutil.Timer
 		defer refreshTimer.Stop()
 		refreshTimer.Reset(refreshTimerDuration / 2)
@@ -1624,7 +1627,7 @@ func (m *Manager) RunBackgroundLeasingTask(ctx context.Context) {
 				m.refreshSomeLeases(ctx, true /*refreshAll*/)
 			case <-refreshTimer.C:
 				refreshTimer.Read = true
-				refreshTimer.Reset(m.storage.jitteredLeaseDuration() / 2)
+				refreshTimer.Reset(getRefreshTimerDuration() / 2)
 
 				// Check for any react to any range feed availability problems, and
 				// if needed refresh the full set of descriptors.

--- a/pkg/sql/catalog/lease/lease_test.go
+++ b/pkg/sql/catalog/lease/lease_test.go
@@ -3674,7 +3674,7 @@ func TestLeaseDescriptorRangeFeedFailure(t *testing.T) {
 						// so the update is detected.
 						if p.Params.ExecutionPhase == scop.PostCommitPhase &&
 							enableAfterStageKnob.Load() &&
-							strings.Contains(p.Statements[0].Statement, "ADD COLUMN") {
+							strings.Contains(p.Statements[0].Statement, "ALTER TABLE t1 ADD COLUMN j INT DEFAULT 64") {
 							rangeFeedResetChan = srv.ApplicationLayer(1).LeaseManager().(*lease.Manager).TestingSetDisableRangeFeedCheckpointFn(true)
 						}
 						return nil
@@ -3684,7 +3684,7 @@ func TestLeaseDescriptorRangeFeedFailure(t *testing.T) {
 						// so the update is detected.
 						if p.Params.ExecutionPhase == scop.PostCommitPhase &&
 							enableAfterStageKnob.Load() &&
-							strings.Contains(p.Statements[0].Statement, "ADD COLUMN") {
+							strings.Contains(p.Statements[0].Statement, "ALTER TABLE t1 ADD COLUMN j INT DEFAULT 64") {
 							<-rangeFeedResetChan
 							srv.ApplicationLayer(1).LeaseManager().(*lease.Manager).TestingSetDisableRangeFeedCheckpointFn(false)
 							enableAfterStageKnob.Swap(false)

